### PR TITLE
DEV: Add implementations for various notification types for the new menu

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Extracted from https://github.com/discourse/discourse/pull/17379.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/user-menu/bookmark-reminder-notification-item.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import UserMenuNotificationItem from "discourse/components/user-menu/notification-item";
+import I18n from "I18n";
+
+export default class UserMenuBookmarkReminderNotificationItem extends UserMenuNotificationItem {
+  get linkTitle() {
+    if (this.notification.data.bookmark_name) {
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/components/user-menu/bookmark-reminder-notification-item.js:6]: Can `notification.data.bookmark_name` ever be blank for a bookmark reminder notification?

Comment (OsamaSayegh) [reply app/assets/javascripts/discourse/app/components/user-menu/bookmark-reminder-notification-item.js:6]: Yes, `bookmark_name` comes from this input field in the bookmark creation modal:



That input field is optional and when it's left blank `bookmark_name` is nil.

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/user-menu/bookmark-reminder-notification-item.js:6]: :+1:

File: app/assets/javascripts/discourse/tests/integration/components/user-menu/granted-badge-notification-item-test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    const template = hbs`<UserMenu::GrantedBadgeNotificationItem @item={{this.notification}}/>`;
+
+    test("links to the badge page and filters by the username", async function (assert) {
+      this.set("notification", getNotification());
+      await render(template);
+      const link = query("li a");
+      assert.ok(link.href.endsWith("/badges/12/tough-guy?username=ossa"));
+    });
+
+    test("the notification's only content is a i18n string", async function (assert) {
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/integration/components/user-menu/granted-badge-notification-item-test.js:48]:
```suggestion
    test("displays the right notification content", async function (assert) {
```

File: app/assets/javascripts/discourse/tests/integration/components/user-menu/group-message-summary-notification-item-test.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+module(
+  "Integration | Component | user-menu | group-message-summary-notification-item",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    const template = hbs`<UserMenu::GroupMessageSummaryNotificationItem @item={{this.notification}}/>`;
+
+    test("the notification displays a simple i18n string", async function (assert) {
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/integration/components/user-menu/group-message-summary-notification-item-test.js:40]:
```suggestion
    test("the notification displays the right content", async function (assert) {
```

**Review by tgxworld — APPROVED** [review]
Some minor comments but the components look good to me. 

Just to confirm, in another PR we'll be using these components right?

**Comment by OsamaSayegh** [comment]
> Just to confirm, in another PR we'll be using these components right?

Oh I missed to include a change that enables these component, I've pushed that change now.

**Pull request merged**